vyper/venom/function.py

```diff
@@ -179,6 +179,25 @@ def chain_basic_blocks(self) -> None:
             else:
                 bb.append_instruction("stop")
 
+    def float_allocas(self):
+        entry_bb = self.entry
+        assert entry_bb.is_terminated
+        tmp = entry_bb.instructions.pop()
+
+        for bb in self.get_basic_blocks():
+            if bb is entry_bb:
+                continue
+
+            # "fast" way to strip allocas from each basic block
+            def is_alloca(inst):
+                return inst.opcode in ("alloca", "palloca")
+
+            bb.instructions.sort(key=is_alloca)
+            while len(bb.instructions) > 0 and is_alloca(bb.instructions[-1]):
+                entry_bb.insert_instruction(bb.instructions.pop())
```

Comment:
```suggestion
            # Extract alloca instructions
            non_alloca_instructions = []
            for inst in bb.instructions:
                if inst.opcode in ("alloca", "palloca"):
                    entry_bb.insert_instruction(inst)
                else:
                    non_alloca_instructions.append(inst)

            # Replace original instructions with filtered list
            bb.instructions = non_alloca_instructions
```

Reply:
this causes a slight regression in CurveStableSwapNG but a slight improvement in VaultV3.vy